packages/rum/src/boot/startRecording.spec.ts

```diff
@@ -37,47 +37,50 @@ describe('startRecording', () => {
     textField = document.createElement('input')
     sandbox.appendChild(textField)
 
-    setupBuilder = setup()
-      .withParentContexts({
-        findView() {
-          return {
-            session: {
-              id: session.getId(),
-            },
-            view: {
-              id: viewId,
-            },
-          }
-        },
-      })
-      .withSession(session)
-      .withConfiguration({
-        defaultPrivacyLevel: DefaultPrivacyLevel.ALLOW,
-      })
-      .beforeBuild(({ lifeCycle, applicationId, configuration, parentContexts, session }) => {
-        const recording = startRecording(
-          lifeCycle,
-          applicationId,
-          configuration,
-          session,
-          parentContexts,
-          getDeflateWorkerSingleton()!
-        )
-        stopRecording = recording ? recording.stop : noop
-        return { stop: stopRecording }
-      })
-
     const requestSendSpy = spyOn(HttpRequest.prototype, 'send')
     ;({
       waitAsyncCalls: waitRequestSendCalls,
       expectNoExtraAsyncCall: expectNoExtraRequestSendCalls,
     } = collectAsyncCalls(requestSendSpy))
+
+    loadDeflateWorkerSingleton((deflateWorker) => {
+      setupBuilder = setup()
+        .withParentContexts({
+          findView() {
+            return {
+              session: {
+                id: session.getId(),
+              },
+              view: {
+                id: viewId,
+              },
+            }
+          },
+        })
+        .withSession(session)
+        .withConfiguration({
+          defaultPrivacyLevel: DefaultPrivacyLevel.ALLOW,
+        })
+        .beforeBuild(({ lifeCycle, applicationId, configuration, parentContexts, session }) => {
+          const recording = startRecording(
+            lifeCycle,
+            applicationId,
+            configuration,
+            session,
+            parentContexts,
+            deflateWorker!
+          )
```

Comment:
what about using `startDeflateWorkerSingleton` to return a worker for the tests that need one?
IMO, no need to add more complexity to the setup